src/html.js

```diff
@@ -125,4 +125,92 @@ const htmlGenerator = function* ({ raw: literals }, ...expressions) {
   }
 };
 
-export { html, htmlGenerator };
+/**
+ * @param {{ raw: string[] }} literals Tagged template literals.
+ * @param {...any} expressions Expressions to interpolate.
+ * @yields {string} The HTML strings.
+ */
+const htmlAsyncGenerator = async function* ({ raw: literals }, ...expressions) {
+  let index = 0;
+
+  for (; index !== expressions.length; ++index) {
+    let literal = literals[index];
+    let expression;
+
+    if (expressions[index] instanceof Promise) {
+      expressions[index] = await expressions[index];
+    }
+
+    if (typeof expressions[index] === "string") {
+      expression = expressions[index];
+    } else if (
+      expressions[index] === undefined ||
+      expressions[index] === null
+    ) {
+      expression = "";
+    } else {
+      if (
+        typeof expressions[index][Symbol.iterator] === "function" ||
+        typeof expressions[index][Symbol.asyncIterator] === "function"
+      ) {
+        const isRaw =
+          literal.length !== 0 && literal.charCodeAt(literal.length - 1) === 33;
+
+        if (isRaw) {
+          literal = literal.slice(0, -1);
+        }
+
+        if (literal.length) {
+          yield literal;
+        }
+
+        for await (const value of expressions[index]) {
+          if (typeof value === "string") {
+            expression = value;
+          } else if (value === undefined || value === null) {
+            continue;
+          } else if (
+            typeof value[Symbol.iterator] === "function" ||
+            typeof value[Symbol.asyncIterator] === "function"
+          ) {
+            expression = "";
+
+            for await (const innerValue of value) {
+              expression += innerValue ?? "";
```

Comment:
You might need a couple of options about escaping here.

Reply:
It uses the same logic, if that whole generator was prefixed with `!`, the whole thing is rendered raw, if not the whole thing is escaped

Also, you might've noticed but for consistency with `Array.join`, we don't take into account `undefined` and `null`

Reply:
And btw, htmlGenerator will never return these types, only string, so these are here for consistency with the outer type handling